File: python/CrawlerLogFolder.py
def minOperations(logs) -> int:
    folder_stack = []

    for current_operation in logs:
        if current_operation == "../":
            # Move up to parent directory if not already at main folder
            if folder_stack:
                folder_stack.pop()
        elif current_operation != "./":
            # Enter a new folder
            folder_stack.append(current_operation)
        # Ignore "./" operations as they don't change the current folder
    # The stack size represents the number of folders deep we are
    return len(folder_stack)
print(minOperations(["d1/","d2/","../","d21/","./"]))

File: JAVA/java/src/CrawlerLogFolder.java
public class CrawlerLogFolder {

    public static int minOperations(String[] logs) {
        int folderDepth = 0;

        // Iterate through each log operation
        for (String currentOperation : logs) {
            // Go up one directory if "../" is encountered, but don't go above the root
            if (currentOperation.equals("../")) {
                folderDepth = Math.max(0, folderDepth - 1);
            }
            // Increase depth if the log is not 'stay in the current directory'
            else if (!currentOperation.equals("./")) {
                // Go down one directory
                folderDepth++;
            }
            // Ignore "./" operations as they don't change the current folder
        }

        return folderDepth;
    }

    public static void main(String[] args) {
        System.out.println(minOperations(new String[] {"d1/","d2/","../","d21/","./"}));


    }
}
